core/modules/replica/ControllerApp.cc

```diff
@@ -434,14 +645,145 @@ int ControllerApp::runImpl() {
             },
             _priority,
             not _doNotTrackRequest);
-    } else if ("SQL" == _request) {
-        ptr = controller->sql(
+    } else if ("INDEX" == _request) {
+        bool const hasTransactions = _transactionId != numeric_limits<uint32_t>::max();
+        ptr = controller->index(
+            _workerName,
+            _sqlDatabase,
+            _chunkNumber,
+            hasTransactions,
+            _transactionId,
+            [&] (IndexRequest::Ptr const& ptr_) {
+                ::printRequest(ptr_,
+                               _indexFileName);
+            },
+            _priority,
+            not _doNotTrackRequest);
+    } else if ("SQL_QUERY" == _request) {
+        ptr = controller->sqlQuery(
             _workerName,
             _sqlQuery,
             _sqlUser,
             _sqlPassword,
             _sqlMaxRows,
-            [&] (SqlRequest::Ptr const& ptr_) {
+            [&] (SqlQueryRequest::Ptr const& ptr_) {
+                ::printRequest(ptr_,
+                               ptr_->responseData(),
+                               ptr_->performance(),
+                               _sqlPageSize);
+            },
+            _priority,
+            not _doNotTrackRequest);
+    } else if ("SQL_CREATE_DATABASE" == _request) {
+        ptr = controller->sqlCreateDb(
+            _workerName,
+            _sqlDatabase,
+            [&] (SqlCreateDbRequest::Ptr const& ptr_) {
+                ::printRequest(ptr_,
+                               ptr_->responseData(),
+                               ptr_->performance(),
+                               _sqlPageSize);
+            },
+            _priority,
+            not _doNotTrackRequest);
+    } else if ("SQL_DELETE_DATABASE" == _request) {
+        ptr = controller->sqlDeleteDb(
+            _workerName,
+            _sqlDatabase,
+            [&] (SqlDeleteDbRequest::Ptr const& ptr_) {
+                ::printRequest(ptr_,
+                               ptr_->responseData(),
+                               ptr_->performance(),
+                               _sqlPageSize);
```

Comment:
all these duplicate printRequest are visually noisy, and you're passing a pointer, and then member data extracted from the pointer. Why not just pass the pointer and use it in the called function. Or as mentioned above, why not make `printRequest` (or data extraction for `printRequest` a member function of the printed class (e.g. in this case `SqlDeleteDbRequest`)?

Reply:
I will consider code refactoring here!

Reply:
...actually, having though more on this issue, I'v decided to go for the refactoring now. The proposal has been registered in my _TODO_ list:
https://confluence.lsstcorp.org/display/DM/Refactor+results+reporting+in+the+implementation+of+class+ControllerApp